### PR TITLE
fix(package): fix incorrect deck count tracking

### PR DESCRIPTION
**Pull request by rikumiyao**
The current zone change logic to handle deck size doesn't handle the case when cards move between player's decks. This fixes that issue.

### CODE REVIEW

**Comment by rikumiyao** [comment]
hold on this until we figure out why the server updates slowly, and probably have test cases too.

**Comment by Lange** [comment]
The `Check for vulnerabilities` step is failing because of the changes to `package-lock.json` in this PR. None of these changes seem relevant to this PR, so I think you can safely remove them.

File: test/index.spec.ts [hunk 1]
```diff
@@ -98,4 +99,49 @@ describe('hearthstone-log-watcher', () => {
 			});
 		});
 	});
+
+	describe('parsing_2', () => {
```
Comment (Lange) [inline test/index.spec.ts]: It is possible to create a more concise test which only tests the thing being changed by this pull request? A lot of this seems like duplication, on the surface.

Comment (rikumiyao) [reply test/index.spec.ts]: We could omit only the part where we compare event counters, but we would still need to read the whole log, unless we want to produce event triggers for our line readers artificially.

Comment (Lange) [reply test/index.spec.ts]: Can you walk me through what this log has which the other log is missing? Maybe the name of this test can be changed from `parse_2` to have a clearer meaning which illustrates what it is testing that the other `parsing` test suite does not test.

Comment (rikumiyao) [reply test/index.spec.ts]: I think the other log is missing a toggwaggle interaction, which is mainly what caused the bug last time

Comment (rikumiyao) [reply test/index.spec.ts]: that should suggest we rename the test `test_togwaggle` or `test_deck_swap`

Comment (Lange) [reply test/index.spec.ts]: I think a name which makes it clear that this test is testing the deck swap functionality of the togwaggle card specifically would be great, even if it has to be a little verbose. Something like `describe('togwaggle deck swap'` would be perfect

**Comment by Lange** [comment]
I have a couple more suggestions to make, but those suggestions depend on https://github.com/Tespa/hearthstone-parser/pull/18

**Comment by Lange** [comment]
These audit errors currently seem to be unfixable. When I fix one, it unfixes another. It's a game of whack-a-mole which I do not know how to win. We'll have to wait a few days until some of these errors get fixed upstream, so that we are less reliant on me manually patching `package-lock.json` by hand.

File: test/index.spec.ts [hunk 2]
```diff
@@ -97,5 +97,25 @@ describe('hearthstone-log-watcher', () => {
 				'tag-change': 852
 			});
 		});
+
+		it('should correctly handle a togwaggle deck swap', () => {
+			const logFilePath = path.join(__dirname, '/artifacts/togwaggle_deck_swap.log');
```
Comment (Lange) [inline test/index.spec.ts]: At what point in this log file does the togwaggle deck swap occur? Is it near the end of the log?

Comment (rikumiyao) [reply test/index.spec.ts]: Near the end I think, 1 or 2 turns before the last turn in the log

Comment (Lange) [reply test/index.spec.ts]: Perfect 👌

**Pull request merged**